src/Facets/DeBridgeFacet.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { IDeBridgeGate } from "../Interfaces/IDeBridgeGate.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+
+/// @title DeBridge Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through DeBridge Protocol
+contract DeBridgeFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Storage ///
+
+    bytes32 internal constant NAMESPACE = keccak256("com.lifi.facets.debridge");
+
+    /// @notice The contract address of the spoke pool on the source chain.
+    IDeBridgeGate private immutable deBridgeGate;
+
+    /// Types ///
+
+    /// @param executionFee Fee paid to the transaction executor.
+    /// @param flags Flags set specific flows for call data execution.
+    /// @param fallbackAddress Receiver of the tokens if the call fails.
+    /// @param data Message/Call data to be passed to the receiver
+    ///             on the destination chain during the external call execution.
+    struct SubmissionAutoParamsTo {
+        uint256 executionFee;
+        uint256 flags;
+        bytes fallbackAddress;
+        bytes data;
+    }
+
+    /// @param permit deadline + signature for approving the spender by signature.
+    /// @param useAssetFee Use assets fee for pay protocol fix (work only for specials token)
+    /// @param nativeFee Native fee for the bridging when useAssetFee is false.
+    /// @param referralCode Referral code.
+    /// @param autoParams Structure that enables passing arbitrary messages and call data.
+    struct DeBridgeData {
+        bytes permit;
```

Comment:
I think you could save some gas by rearranging the parameters (cause in the struct they will be tightly packed).
I think the uint256 parameter should be first and then the other...i dont know in which order exactly, would recommend to try...